src/handler.rs

```diff
@@ -577,6 +600,237 @@ where
 
         Ok(())
     }
+
+    fn get_inflight_fd(
+        &mut self,
+        inflight: &vhost::vhost_user::message::VhostUserInflight,
+    ) -> VhostUserResult<(vhost::vhost_user::message::VhostUserInflight, File)> {
+        // Total size of the inflight queue region
+        let total_mmap_size =
+            self.get_inflight_queue_size(inflight.queue_size) * inflight.num_queues as usize;
+
+        // Create a memfd region to hold the queues for inflight I/O tracking
+        let (dup_file, total_mmap_size) = self.memfd_alloc(total_mmap_size)?;
+
+        let dup_inflight_file = match dup_file {
+            Some(f) => f,
+            None => return Err(VhostUserError::SlaveInternalError),
+        };
+
+        Ok((
+            VhostUserInflight {
+                mmap_size: total_mmap_size as u64,
+                mmap_offset: 0,
+                num_queues: inflight.num_queues,
+                queue_size: inflight.queue_size,
+            },
+            dup_inflight_file,
+        ))
+    }
+
+    fn set_inflight_fd(
+        &mut self,
+        inflight: &vhost::vhost_user::message::VhostUserInflight,
+        file: File,
+    ) -> VhostUserResult<()> {
+        let ret_val = -1;
+
+        // Need to unmap any previously mmaped regions as closing the
+        // associated file doesn't unmap it automatically.
+        // unsafe as extern munmap is called
+        if let Some(inflight_addr) = self.inflight_state.get_inflight_mapping_addr() {
```

Comment:
The mapping address should be reset to NULL after unmapped, otherwise the address may get unmapped multiple times.